gm8emulator/src/game.rs

```diff
@@ -2211,16 +2241,55 @@ impl Game {
             self.renderer.upload_sprite(Box::new([0, 0, 0, 0]), 1, 1, 0, 0).expect("Failed to upload blank sprite");
         }
 
-        for ev in replay.startup_events.iter() {
-            self.stored_events.push_back(ev.clone());
+        let mut clean_state = true;
+        if start_save_path.is_some() {
+            let mut save_buffer = savestate::Buffer::new();
+            match SaveState::from_file(start_save_path.unwrap(), &mut save_buffer) {
+                Ok(state) => {
+                    let (rep, ren) = state.clone().load_into(&mut self);
+                    if !replay.contains_part(&rep) {
+                        panic!("Savestate is not part of replay");
+                    }
+
+                    frame_count = rep.frame_count();
+                    clean_state = state.clean_state;
+                    self.renderer.set_state(&ren);
+                },
+                Err(e) => {
+                    panic!("(Fatal) Error loading savestate file: {:?}", e);
+                }
+            }
+        } else {
+            for ev in replay.startup_events.iter() {
+                self.stored_events.push_back(ev.clone());
+            }
+            self.init()?;
+            handle_scene_change!(self);
         }
-        self.init()?;
-        handle_scene_change!(self);
 
         let mut time_now = Instant::now();
         loop {
             self.window.poll_events();
             self.input.mouse_step();
+            
+            if self.frame_limit_at > 0 && frame_count == self.frame_limit_at || frame_count == replay.frame_count() {
+                if let Some(bin) = &output_bin {
+                    if start_save_path.is_some() {
+                        // Store the current framebuffer since it's used by the savestate. Only matters if there already is a framebuffer stored which is the case when loading a savestate.
+                        self.renderer.resize_framebuffer(self.renderer.stored_size().0, self.renderer.stored_size().1, true);
```

Comment:
```suggestion
                        // Store the current framebuffer since it's used by the savestate.
                        // Only matters if there already is a framebuffer stored which is the case when loading a savestate.
                        self.renderer.resize_framebuffer(self.renderer.stored_size().0, self.renderer.stored_size().1, true);
```